source/background.ts

```diff
@@ -42,3 +42,16 @@ browser.runtime.onInstalled.addListener(async ({reason}) => {
 
 // GitHub Enterprise support
 addDomainPermissionToggle();
+
+// `background` fetch required to avoid avoid CORB introduced in Chrome 73 https://chromestatus.com/feature/5629709824032768
+browser.runtime.onMessage.addListener(({action, payload}, _sender, sendResponse): void | boolean => {
+	if (action === 'fetch') {
+		void fetch(payload)
+			.then(response => sendResponse(response.json()))
+			.catch(error => ({
+				error: error.message // Make error JSON.stringify-able
+			}))
+			.then(sendResponse);
```

Comment:
Thanks to the promised `browser.*` API, you can just return the promise instead of all of this. Drop as much code as possible. 

Example: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage